tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -114,6 +114,72 @@ public void Merge_should_add_expected_stage()
             stages[0].Should().Be("{ $merge : { into : { db : 'database', coll : 'collection' } } }");
         }
 
+        [Fact]
+        public void UnionWith_should_add_expected_stage()
+        {
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>();
+            var withCollection = Mock.Of<IMongoCollection<BsonDocument>>(
+                coll => coll.CollectionNamespace == CollectionNamespace.FromFullName("db.test"));
+
+            var unionWithPipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .AppendStage<BsonDocument, BsonDocument, BsonDocument>("{ $match : { b : 1 } }");
+            var result = pipeline.UnionWith<BsonDocument, BsonDocument, BsonDocument>(
+                withCollection,
+                unionWithPipeline);
```

Comment:
With the right intermediate variables the "Act" line can almost always be a one liner:

    var pipeline = new EmptyPipelineDefinition<BsonDocument>();
    var withCollection = Mock.Of<IMongoCollection<BsonDocument>>(
        coll => coll.CollectionNamespace == CollectionNamespace.FromFullName("db.test"));
    var withPipeline = new EmptyPipelineDefinition<BsonDocument>()
        .AppendStage<BsonDocument, BsonDocument, BsonDocument>("{ $match : { b : 1 } }");

    var result = pipeline.UnionWith(withCollection, withPipeline);

    var stages = RenderStages(result, BsonDocumentSerializer.Instance);
    stages[0].Should().Be("{ $unionWith : { coll : 'test', pipeline : [{ $match : { b : 1 } }] } }");